### PR TITLE
Allow AWS VMs to share the same VPCs with previous runs.

### PR DESCRIPTION
**Pull request by stfeng**

### CODE REVIEW

**Comment by stfeng** [comment]
@ehankland 

Hey Eric, this is the change I was talking about that allows a user to share a VPC across multiple PKB runs.

I changed AWS related stuff at this moment, want to let you take a look at this first. If this is on the right direction, I will change other providers (Azure)

**Comment by stfeng** [comment]
test this please.

File: perfkitbenchmarker/aws/aws_virtual_machine.py [hunk 1]
```diff
@@ -36,6 +36,8 @@
 
 FLAGS = flags.FLAGS
 
+flags.DEFINE_string(
+    'existing_subnet_id', None, 'An existing subnet to create a VM in.')
```
Comment (ehankland) [inline perfkitbenchmarker/aws/aws_virtual_machine.py]: Please also document that placement groups wont be used if this flag is used

File: perfkitbenchmarker/benchmark_spec.py [hunk 2]
```diff
@@ -321,8 +321,11 @@ def PrepareVm(self, vm, firewall):
     vm.Create()
     logging.info('VM: %s', vm.ip_address)
     logging.info('Waiting for boot completion.')
-    for port in vm.remote_access_ports:
-      firewall.AllowPort(vm, port)
+    # If the user has requested to use a specific subnet ID, that has already
+    # been setup with allowed port, we cannot invoke it again here.
+    if FLAGS.existing_subnet_id is None:
```
Comment (ehankland) [inline perfkitbenchmarker/benchmark_spec.py]: I don't like checking a cloud specific flag in a file shared by many clouds - can you pull it up into aws_network.py?

**Comment by voellm** [comment]
I plan to roll the next release today.  Is this ready to integrate?

**Comment by stfeng** [comment]
@voellm: I found my current way of doing this quite hacky. I think a proper way of sharing resources across multiple PKB runs is a manifest file based approach, similar to static VM sharing. I'm going to close this PR and chat with Eric more about this.

**Pull request closed without merging**